numpy/lib/recfunctions.py

```diff
@@ -811,6 +811,285 @@ def repack_fields(a, align=False, recurse=False):
     dt = np.dtype(fieldinfo, align=align)
     return np.dtype((a.type, dt))
 
+def _get_fields_and_offsets(dt, offset=0):
+    """
+    Returns a flat list of (name, dtype, count, offset) tuples of all the
+    scalar fields in the dtype "dt", including nested fields, in left
+    to right order.
+    """
+    fields = []
+    for name in dt.names:
+        field = dt.fields[name]
+        if field[0].names is None:
+            count = 1
+            for size in field[0].shape:
+                count *= size
+            fields.append((name, field[0], count, field[1] + offset))
+        else:
+            fields.extend(_get_fields_and_offsets(field[0], field[1] + offset))
+    return fields
+
+def structured_to_unstructured(arr, dtype=None):
```

Comment:
I'd prefer an implementation that just does `arr.astype(replace_all_fields(arr.dtype, dtype)).view(dtype)`, since then it could take a `copy` parameter to forward to `astype`

Reply:
Good idea, done. 

As a bonus, the function now returns a view instead of a copy if all the fields have the same dtype, which is the use-case people said they used on the mailing list.